### PR TITLE
docs(root): improve README.md documentation clarity

### PR DESCRIPTION
**Pull request by jeremylenz**
Provide a better explanation of what vendor is
 improve organization; fix typos

## Updated Packages

* [x] root
* [ ] @theforeman/builder
* [ ] @theforeman/env
* [ ] @theforeman/eslint-plugin-foreman
* [ ] @theforeman/vendor
* [ ] @theforeman/vendor-dev
* [ ] @theforeman/vendor-core

## PR Type

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (whitespace, formatting, missing semicolons, etc.)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [x] Documentation content changes
* [ ] Other… Please describe:

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Does this PR fixes open issues?

* [ ] Yes
* [x] No

## Do you use this PR in another repository?

* [ ] Yes
* [x] No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+* See [discussion.](https://community.theforeman.org/t/redesigning-the-javascript-stack/13470)
+* Watch the [deep dive](https://youtu.be/uO-YTjl-si0) explaining the redesign.
+
+## Vendor packages
+
+Foreman and its plugins use many of the same JavaScript dependencies (npm packages).  The main purpose of `vendor` is to __gather these common dependencies into a single package.__
+
+- [@theforeman/vendor-core](packages/vendor-core) - This is the actual list of common dependencies.  Is a dependency of `vendor` and `vendor-dev`.
+- [@theforeman/vendor](packages/vendor) - The dependencies in `vendor-core`, bundled to be used in Webpack.
+- [@theforeman/vendor-dev](packages/vendor-dev) - The dependencies in `vendor-core`, unbundled for use in development.
```
Comment (waldenraines) [inline README.md]: It may be useful in this section to provide an example of how you would import these from a plugin

Comment (jeremylenz) [reply README.md]: @waldenraines Updated with note

File: README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+_Note:_ In Foreman & plugins code, you __never__ need to import anything explicitly from vendor (`import react from '@theforeman/vendor'`); you can continue to import all modules like normal (`import react from 'react';`).
```
Comment (sharvit) [inline README.md]: It is true except for one exception `scss`.
See: https://github.com/theforeman/foreman-js/tree/master/packages/vendor#stylesheets

Does it need more clarification?

Comment (jeremylenz) [reply README.md]: @sharvit thanks!  Updated with another note

**Comment by jeremylenz** [comment]
Also rebased to latest master

**Pull request merged**